apps/docs/pages/guides/auth/auth-postgres-hooks.mdx

```diff
@@ -0,0 +1,851 @@
+import Layout from '~/layouts/DefaultGuideLayout'
+import {CodeBlock, Admonition, Accordion} from 'ui'
+import Image from 'next/image'
+
+export const meta = {
+  id: 'auth-postgres-hooks',
+  title: 'Auth Hooks (Postgres)',
+  description: 'Use Supabase Postgres Functions to customize your authentication flow',
+}
+
+<Admonition type="info">
+Hooks are in beta. They will be available for all projects in early 2024.
+</Admonition>
+
+
+Supabase allows you to use PostgreSQL functions to alter the default Supabase Auth flow. Developers can use hooks to add custom behavior that's not supported natively.
+
+Hooks help you:
+
+- Track the origin of user signups by adding metadata
+- Improve security by adding additional checks to password and multi-factor authentication
+- Support legacy systems by integrating with identity credentials from external authentication systems
+- Add additional custom claims to your JWT
+
+
+You can use hooks at specific points along an Auth flow to perform custom behavior via Postgres Functions. These are known as **extensibility points**. The following extensibility points are available:
+
+| Extensibility Point              | Available on Plan       |
+|----------------------------------|-------------------------|
+| MFA Verification Attempt         | Teams and Enterprise    |
+| Password Verification Attempt    | Teams and Enterprise    |
+| Custom Access Token              | Free                    |
+
+You can link a hook to an extensiblity point to signal to Supabase Auth that it should make use of the hook.
+
+
+Below is a flow chart illustrating how Supabase Auth uses an Auth Hook:
+
+<Image
+  alt="Auth Hooks Flowchart"
+  width={'1000'}
+  height={'400'}
+  src="/docs/img/guides/auth-postgres-hooks/hooks-flowchart.png"/>
+
+
+## Create a Hook
+### Write a Hook
+
+In the dashboard, go to `SQL Editor > Templates` and select `Add Auth Hook (General)`. The function for your hook should look like this:
+
+```sql
+create or replace function
+mfa_verification_attempt(event jsonb) returns jsonb as $$
+declare
+  -- Add your variables here
+begin
+  -- Add your logic here
+  return response
+end;
+$$ language plpgsql;
+```
+
+Note that the JSONB operator `?` can't be used when writing a hook, because it is a reserved character in the ORM used by Supabase Auth.
+
+### Handle Hook errors
+
+Unless otherwise specified, your hook should return errors with the following fields:
```

Comment:
Avoid language like "unless" ... 

This is it, this is the spec, it's talking about concrete auth hooks.

You should also add some sentences here to talk about what does an "error" mean in hook terms. It's a runtime error that modifies the behavior of Supabase Auth; not a developer error such as division by zero.

```suggestion
To return an error from a hook, return a JSON object with the following properties:
```